### ISSUE TITLE
Xamarin UWP access to a SQLite Database on a Network Drive

### ISSUE THREAD

**Issue by KevinConManApp** [praeclarum/sqlite-net#1073]
Hi There,
This may be more of a Xamarin:UWP issue but as it relates to a SQLite connection I am trying here as well.
I have developed a Xamarin Forms App that accesses a SQLite Database via sqlite-net-pcl. For the Android and iOS apps, I copy a centralised database file to the local device and sync it back to the central one as required.
However, for the UWP app I am connecting directly to the centralised one. It sits in Azure File Storage that is mapped as a network drive - W: on the Win10 Device that the UWP app runs on.
While this connects fine when running the app in VS2019, when I deploy a UWP package to another win10 device, I continually get a connection error that appears to be related to access permissions.
I have followed advice regarding 'rescap: broadFileSystemAccesss' and can see the database as a Windows.Storage file but suspect it may relate to being denied System.IO access. But do not understand why I can connect in Debug mode, which also works when I debug to a remote Win10 device.
Any ideas would be most welcome.
Regards,
Kevin.

**Comment by NathanKOMet** [comment]
Did you ever find a solution to this?

**Comment by KevinConManApp** [comment]
Yes, but only by allowing the Developer Mode setting on the Win10 device. This works for us in that the UWP app is not downloaded from a store rather it is installed in-house along with some other VSTO Addins that combine to provide the solution we need.

**Comment by NathanKOMet** [comment]
Hmm. I have done that. I've no idea what the issue is. It works when running in VS2022 but not when installed on another machine. I tried installing it on the machine I develop on and it doesn't work there either, even though it'll work when running in VS. Were there any other steps that you had taken that might be able to help me out?

**Comment by NathanKOMet** [comment]
For reference I wrote this as a test just to see try and narrow down what the issue is.
It throws "SQLite Error 14: 'Unable to open database file'" at db.Open()
`string entry = "";
string dbpath = @"S:\TestDB.db");
using (SqliteConnection db = new SqliteConnection($"Filename={dbpath}"))
{
    db.Open();
    string textCommand = "SELECT Source from Materials WHERE Material = @keyValue";
    SqliteCommand selectCommand = new SqliteCommand(textCommand, db);
    selectCommand.Parameters.AddWithValue("@keyValue", "PMMA");
    SqliteDataReader query = selectCommand.ExecuteReader();
    while (query.Read()) { entry = query.GetString(0); }
    db.Close();
}
return entry;`

**Comment by KevinConManApp** [comment]
My rescap settings in the manifest are:
    <rescap:Capability Name="broadFileSystemAccess" />
    <rescap:Capability Name="developmentModeNetwork" />

**Comment by NathanKOMet** [comment]
Just had the opportunity to test this out and I think you've provided the solution for me.

The first test I did actually worked. I don't know why but I didn't have developmentModeNetwork on for some reason. I'll have to test it on main function tomorrow, but it looks like you've fixed this. This has been bugging me for a while now, so you've really made my day. Thanks so much :)

**Comment by KevinConManApp** [comment]
All good, Nathan. What goes around comes around.
